.github/workflows/ci-actions-incremental.yml

```
@@ -829,12 +829,22 @@ jobs:
           build-scan-capture-strategy: ON_DEMAND
           job-name: "Quickstarts Compilation - JDK ${{matrix.java.name}}"
           wrapper-init: true
+      - uses: actions/github-script@v7
+        id: get-quickstarts-branch
+        with:
+          result-encoding: string
+          script: |
+            const branch = process.env.GITHUB_BASE_REF ? process.env.GITHUB_BASE_REF : process.env.GITHUB_REF_NAME
+            if (branch == 'main') {
+              return 'development'
+            } else {
+              return branch
+            }
       - name: Compile Quickstarts
         env:
           CAPTURE_BUILD_SCAN: true
         run: |
-          git clone https://github.com/quarkusio/quarkus-quickstarts.git && cd quarkus-quickstarts
-          git checkout development
+          git clone -b ${{ steps.get-quickstarts-branch.outputs.result }} https://github.com/quarkusio/quarkus-quickstarts.git && cd quarkus-quickstarts
```

Use a shallow clone to speed up the cloning process: 

```suggestion
          git clone --depth=1 -b ${{ steps.get-quickstarts-branch.outputs.result }} https://github.com/quarkusio/quarkus-quickstarts.git && cd quarkus-quickstarts
```